src/vim-loader/vimScene.ts

```diff
@@ -17,40 +18,67 @@ export class VimScene {
     this.vim = vim
     this.geometry = geometry
     this.geometryBuilder = geometryBuilder
-    this.elementToNodes = this.mapElements()
+    this.elementIndexToNodeIndices = this.mapElementIndexToNodeIndices()
+    this.elementIdToIndex = this.mapElementIdToIndex()
   }
 
-  private mapElements (): Map<number, number[]> {
-    const nodeCount = this.geometry.getNodeCount()
+  private mapElementIndexToNodeIndices (): Map<number, number[]> {
     const map = new Map<number, number[]>()
-    for (let i = 0; i < nodeCount; i++) {
-      const elementId = this.getElementIdFromNodeIndex(i)
-      if (elementId === undefined) continue
+    const nodeElements = this.getElementTable(this.getNodeTable())
+    const nodeCount = nodeElements.length
 
-      const nodes = map.get(elementId)
+    for (let node = 0; node < nodeCount; node++) {
+      const element = nodeElements[node]
+      if (element === undefined) continue
+
+      const nodes = map.get(element)
       if (nodes) {
-        nodes.push(i)
+        nodes.push(node)
       } else {
-        map.set(elementId, [i])
+        map.set(element, [node])
       }
     }
     return map
   }
 
+  private mapElementIdToIndex (): Map<number, number> {
```

Comment:
Minor: could we clarify the name as `mapElementIdToElementIndex`? The term `Index` here is ambiguous to me as a reader; is it referring to a node index or an element index?